spec.bs

```diff
@@ -0,0 +1,369 @@
+
+        then:
+          1. [=Queue a global task=] on the <dfn>browsing topics task source</dfn> given [=this=]'s [=relevant global object=] to [=reject=] |p| with a "{{NotAllowedError}}" {{DOMException}}.
+          1. Abort these steps.
+    1. Let |result| be the result from running the [=topics for caller=] algorithm, with |C| as input.
+    1. If options.observe is true:
+        1. Run the [=collect page topics calculation input data=] algorithm with |C|.page as input.
+        1. Run the [=collect topics caller origin=] algorithm with |C|.page and |C|.caller-origin as input.
+    1. [=Queue a global task=] on the [=browsing topics task source=] given [=this=]'s [=relevant global object=] to perform the following steps:
+        1. [=Resolve=] |p| with |result|.
+  1. Return |p|.
+
+</section>
+
+<section>
+  <h2 id="http-request-header-header">HTTP Request Header</h2>
+
+  Topics can be send in the HTTP header for navigation and fetch() request. The corresponding response header is able to specify whether the caller should to be recorded.
+
+  <h3 id="topics-allowable-http-request">Topics-allowable HTTP request</h3>
+
+  <h4 id="topics-allowable-navigation-request-header">Topics-allowable navigation request</h4>
+  The following algorithm determines whether topics are allowed to be sent on a navigation request header:
+    1. Determine the [=topics-caller-context=] |C|.
+    1. If |C|.caller-origin is not <a href="https://www.w3.org/TR/secure-contexts/#potentially-trustworthy-origin">potentially trustworthy</a>:
+        1. Return false.
+    1. If the navigation happens on an iframe:
+        1. If the iframe's embeding document is not in [=secure context=]:
+            1. Return false.
+        1. For each feature |F| in {"browsing-topic", "interest-cohort"}:
+            1. Run the <a href="https://www.w3.org/TR/permissions-policy-1/#define-inherited-policy-in-container">inherited policy for feature in container at origin</a> algorithm with input arguments |feature|: |F|, |origin|: the navigation URL origin, <var ignore=''>container</var>: the iframe container. If the algorithm returns "Disabled":
+                1. Return fasle.
+    1. If the user preference setting disallows the access to topics from |C|.caller- origin and |C|.page.top-level-origin:
+        1. Return false.
+    1. Return true.
+
+  Note: In Chrome's experimentation phase, top-level navigation won't be topics-allowable, and subframe navigation will additioanlly require a valid <a href="https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/explainer.md">Origin Trial</a> token to exist in the iframe's container document for the request to be topics-allowable.
+
+  <h4 id="topics-allowable-fetch-request-header">Topics-allowable fetch() request</h4>
+  The following algorithm determines whether topics are allowed to be sent on a fetch() request header:
```

Comment:
Nothing calls this algorithm. You need to specify the modifications to the HTML Standard's navigation sections that you would be making, which call this algorithm and (presumably) append a header to a specific response.